spec/features/home_page_spec.rb

```diff
@@ -0,0 +1,93 @@
+feature 'viewing home' do
+  scenario 'welcome message' do
+    visit('/')
+    expect(page).to have_content "Welcome to Chitter!"
+  end
+
+  scenario 'sign up button' do
+    visit('/')
+    click_button('Sign up here...')
+    expect(page).to have_content "Ready to join the Chitter Family?"
+  end
+
+  scenario 'user cannot add a peep to the page before signing up/logging in' do
+    visit('/')
+    expect(page).to have_no_button('Peep it!')
+  end
+
+  scenario 'log in button' do
+    visit('/')
+    click_button('Log in here...')
+    expect(page).to have_content "Enter your username and password below:"
+  end
+  
+  scenario 'user can add a peep to the page after logging in' do
+    User.create(username: 'Sarahness', name: 'Sarah Smith', email: 'email@email.com', password: 'pwpwpwpw')
+    
+    visit('/log_in')
```

Comment:
Could you use spec_helper methods more to make these tests read more easily, be shorter, avoid repetition?

Reply:
Yes, I should have done that, although some of the Sarahs are subtly different to each other

Reply:
The Sarahs did turn out to be all the same, but trying to add it in after the wipe table method in the rspec before config didn't work - I remember now that I also had issues making the database_helper methods use the DatabaseConnection class methods. It seems that the connection to the database isn't actually yet made at that point, and only when the tests actually happen? Would you be able to shed any light?

Reply:
Hey! An example of doing this is here (link below), used on the Battle pair challenge. You can find some guidance on it in the relevant section of the challenge.

I should have been clearer: the concept is to create a file called web_helper.rb inside spec/features which just has some methods which run the repetitive code. You require this file where you have used the repetitive code and replace the code with the declarative method names stored in the helper file to be DRYer. Having the file in spec is ok because as long as you don't include spec in the name, Rspec will not think it is a test itself.

https://github.com/almorcrette/Battle/blob/main/spec/features/web_helpers.rb